LibreNMS/Snmptrap/Handlers/LinkUp.php

```diff
@@ -52,20 +52,8 @@ public function handle(Device $device, Trap $trap)
             return;
         }
 
-        $trapOperStatus = $trap->getOidData("IF-MIB::ifOperStatus.$ifIndex");
-        if ($trapOperStatus) {
-            $port->ifOperStatus = $trapOperStatus;
-        } else {
-            $port->ifOperStatus = 'up';
-        }
-
-        $trapAdminStatus = $trap->getOidData("IF-MIB::ifAdminStatus.$ifIndex");
-        if ($trapAdminStatus) {
-            $port->ifAdminStatus = $trapAdminStatus;
-        } else {
-            // If we receive LinkUp trap, we can safely assume that the ifAdminStatus is also up.
-            $port->ifAdminStatus = 'up';
-        }
+        $trapOperStatus = $trap->getOidData("IF-MIB::ifOperStatus.$ifIndex") ?: 'up';
+        $trapAdminStatus = $trap->getOidData("IF-MIB::ifAdminStatus.$ifIndex") ?: 'up'; // If we receive LinkUp trap, we can safely assume that the ifAdminStatus is also up.
```

Comment:
You don't update $port->ifXxxxStatus with this change. This is a problem because the port Model won't be updated.

Reply:
oops :D

Reply:
```
$port->ifOperStatus = $trap->getOidData("IF-MIB::ifOperStatus.$ifIndex") ?: 'up';
$port->ifAdminStatus = $trap->getOidData("IF-MIB::ifAdminStatus.$ifIndex") ?: 'up';
```

Reply:
The web editor isn't always the best to show you mistakes like that ;)

Reply:
Yep, that is what I intended it to be.